hazelcast/src/main/java/com/hazelcast/map/merge/MergePolicyProvider.java

```diff
@@ -44,8 +42,7 @@ public MapMergePolicy createNew(String className) {
             try {
                 return newInstance(nodeEngine.getConfigClassLoader(), className);
             } catch (Exception e) {
-                nodeEngine.getLogger(getClass()).severe(e);
-                throw rethrow(e);
+                throw new InvalidConfigurationException("Invalid MapMergePolicy: " + className, e);
```

Comment:
If there is an exception, it is now first triggered by the `MergePolicyValidator`, then catched and currently rethrown as `ConfigurationException`. For example:
```java
        try {
            Object mergePolicyInstance = mergePolicyProvider.getMergePolicy(mergePolicyClassname);
            checkMergePolicy(mergeTypeProvider, mergePolicyInstance);
        } catch (InvalidConfigurationException e) {
            throw new ConfigurationException("Merge policy must be an instance of SplitBrainMergePolicy"
                    + " or CacheMergePolicy, but was " + mergePolicyClassname);
        }
```
Before that exception should just be triggered during the merge process, which doesn't throw anything at the user. So I think we're free to handle this like we think it's best.

BTW, I'm still unsure about `InvalidConfigurationException` vs. `ConfigurationException`

Reply:
I switched to `InvalidConfigurationException`, since it seems to be more frequently used by now (60 vs. 28 constructor calls with this PR), although `ConfigurationException` is older.